### ISSUE TITLE
SystemMonitor: Performance tab hangs SystemMonitor when multiple SystemMonitors running

### ISSUE THREAD

**Issue by bcoles** [SerenityOS/serenity#13542]
`SystemMonitor` stops responding when more than one instance is run concurrently.

Navigating to the `Performance` tab seems to be required to reproduce this issue. The `SystemMonitor` does not unfreeze once the second instance is closed.

To reproduce, double click one of the status graphs in the task bar. Usually two instances is sufficient, but if not, repeat until hang.

![SystemMonitor](https://user-images.githubusercontent.com/434827/161990379-bd3f213b-fc79-4345-83c8-e732c9bc06d1.png)

https://github.com/SerenityOS/serenity/issues/13505#issuecomment-1090299354

**Comment by AtkinsSJ** [comment]
It's not just opening a second SystemMonitor instance - launching any GUI app will cause the freeze.

@Lubrsi bisected it to this commit:
https://github.com/SerenityOS/serenity/commit/0a61b45b646ad60de110db25f61f25ba7715aa71

cc: @kleinesfilmroellchen

**Comment by kleinesfilmroellchen** [comment]
That bisect looks very sus, the commit didn't touch the performance tab at all. Can you check if it reproduces with the (not necessarily related) fix in #13540 ?

**Comment by bcoles** [comment]
> That bisect looks very sus, the commit didn't touch the performance tab at all. Can you check if it reproduces with the (not necessarily related) fix in #13540 ?

One step ahead. Already tested with and without that patch. Issue is still present.

**Comment by bcoles** [comment]
Also hit this (once) with #13540 applied:

```
[SystemMonitor(39:40)]: ASSERTION FAILED: list.contains(current_thread)
[SystemMonitor(39:40)]: ../../Kernel/Locking/Mutex.cpp:221 in Kernel::Mutex::block(Kernel::Thread&, Kernel::Mutex::Mode, Kernel::SpinlockLocker<Kernel::Spinlock>&, u32)::<lambda(auto:37&)> [with auto:37 = Kernel::Mutex::BlockedThreadLists]
[SystemMonitor(39:40)]: KERNEL PANIC! :^(
[SystemMonitor(39:40)]: Aborted
[SystemMonitor(39:40)]: at ../../Kernel/Arch/x86/common/CPU.cpp:35 in void abort()
[SystemMonitor(39:40)]: Kernel + 0x00cbc991  Kernel::__panic(char const*, unsigned int, char const*) +0xf1
[SystemMonitor(39:40)]: Kernel + 0x01066f39  abort.localalias +0x245
[SystemMonitor(39:40)]: Kernel + 0x01066cf4  abort.localalias +0x0
[SystemMonitor(39:40)]: Kernel + 0x00b6044f  Kernel::Mutex::block(Kernel::Thread&, Kernel::LockMode, Kernel::SpinlockLocker<Kernel::Spinlock>&, unsigned int) [clone .localalias] +0xb9f
[SystemMonitor(39:40)]: Kernel + 0x00b61842  Kernel::Mutex::lock(Kernel::LockMode, Kernel::LockLocation const&) +0x3b2
[SystemMonitor(39:40)]: Kernel + 0x00806ae3  Kernel::Ext2FSInode::metadata() const [clone .localalias] +0x93
[SystemMonitor(39:40)]: Kernel + 0x009e3464  Kernel::VirtualFileSystem::resolve_path_without_veil(AK::StringView, Kernel::Custody&, AK::RefPtr<Kernel::Custody, AK::RefPtrTraits<Kernel::Custody> >*, int, int) [clone .localalias] +0x26c4
[SystemMonitor(39:40)]: Kernel + 0x009e6dd5  Kernel::VirtualFileSystem::resolve_path(AK::StringView, Kernel::Custody&, AK::RefPtr<Kernel::Custody, AK::RefPtrTraits<Kernel::Custody> >*, int, int) [clone .localalias] +0x55
[SystemMonitor(39:40)]: Kernel + 0x00a010b7  Kernel::VirtualFileSystem::open(AK::StringView, int, unsigned short, Kernel::Custody&, AK::Optional<Kernel::UidAndGid>) +0x87
[SystemMonitor(39:40)]: Kernel + 0x00e74e3e  Kernel::Process::sys$open(AK::Userspace<Kernel::Syscall::SC_open_params const*>) +0x101e
[SystemMonitor(39:40)]: Kernel + 0x00d96153  syscall_handler +0xea3
[SystemMonitor(39:40)]: Kernel + 0x00d952a1  syscall_asm_entry +0x31
```

**Comment by kleinesfilmroellchen** [comment]
Have it tracked down to `fîll_path`. 🤔

**Comment by kleinesfilmroellchen** [comment]
Have found it, just an initialization issue. PR incoming

**Comment by kleinesfilmroellchen** [comment]
Fixed in #13540
